src/main.c

```diff
@@ -275,7 +276,37 @@ static void MX_TIM3_Init(void)
   */
 static void MX_GPIO_Init(void)
 {
-  // TODO - implement GPIO initializations here...
+
+  GPIO_InitTypeDef GPIO_InitStruct = {0};
+
+  /* GPIO Ports Clock Enable */
+  __HAL_RCC_GPIOA_CLK_ENABLE();
+  __HAL_RCC_GPIOB_CLK_ENABLE();
+
+  /* Set default GPIO pin Output Level */
+  HAL_GPIO_WritePin(INSP_VALVE_PORT, INSP_VALVE_PIN, GPIO_PIN_RESET);
+  HAL_GPIO_WritePin(EXP_VALVE_PORT, EXP_VALVE_PIN, GPIO_PIN_RESET);
+  HAL_GPIO_WritePin(TANK_VALVE_PORT, TANK_VALVE_PIN, GPIO_PIN_RESET);
+
+  /* Configure each GPIO pin */
```

Comment:
Is it possible to put all this configuration code in `configuration_private.h`? Is it required to be in `main`?

Reply:
Done.

The initialization for GPIO is now in `platform/init.h` and `platform/init.c`. I haven't moved other initialization (e.g., timers) to this file yet. We can do this in another issue/PR. I want to keep this focused as much as possible on valve management.

I have made #61 for this purpose. Can be picked up next.